### PR TITLE
KHP integration transformation logic: added mappers and some KHP mapping 

### PR DESCRIPTION
**Pull request by GPaoloni**
## Description
This PR:
- Adds to the resources import lambda the script code (from PoC) to map an external resource to an Aselo one, with slight tweeks.
- Unit tests for the above.
- Updates the mappings for KHP resources.

What's left:
- Wire the mapper to the actual Lambda function handler.

### Checklist
- [x] Corresponding issue has been opened](https://tech-matters.atlassian.net/browse/CHI-1774)
- [X] New tests added

### Verification steps
- Unit tests should cover most of the basics for the transformations functions. Confirming that those are testing the intended behavior is much appreciated.
- Manual revision that I'm doing the appropriate mappings to the KHP resources as they come from the staging API.

### CODE REVIEW

File: resources-domain/resources-import-producer/transformExternalResourceToApiResource.ts [hunk 1]
```diff
@@ -0,0 +1,212 @@
...
+      } else if (isTranslatableAttributeMapping(mapping)) {
+        pushTranslatableAttributeMapping({ aseloResource, mapping, context });
+      } else if (isInlineAttributeMapping(mapping)) {
+        pushInlineAttributeMapping({ aseloResource, mapping, context });
+      }
+
+      // Recurse on the children node(s) if any
+      if (children) {
+        // eslint-disable-next-line @typescript-eslint/naming-convention
+        const { _id, objectId, ...rest } = children;
```
Comment (GPaoloni) [inline resources-domain/resources-import-producer/transformExternalResourceToApiResource.ts:175]: Do we want to use `objectId` annd/or `_id`? I was filtering them out here, as they are sibling of languages sometimes (objects like `{ en: ..., fr: ..., _id: 123 }`), but this needs to be removed if we want to make use of them as identifiers.

File: resources-domain/resources-import-producer/tests/unit/transformExternalResourceToApiResource.test.ts [hunk 2]
```diff
@@ -0,0 +1,898 @@
...
+        ],
+      },
+    });
+
+    const result = transformExternalResourceToApiResource(mapping, resource);
+    expect(result).toMatchObject(expected);
+  });
+});
+
+describe('transformExternalResourceToApiResource - Simple mapping, flat structure', () => {
```
Comment (stephenhand) [inline resources-domain/resources-import-producer/tests/unit/transformExternalResourceToApiResource.test.ts:139]: Can we try to use an Arrange Act Assert type naming for tests here in line with our standards, I appreciate in might make the tests a little more verbose but I think we'll appreciate the clarity dowen the road when nobody has looked at this code for 6 month :-P

Also - it looks like you cover most of the mapping functionality, but perhaps some thinks like custom mappers for various attributes could have their own specific tests for that functionality, again, just to be super explicit

File: resources-domain/resources-import-producer/mappings/khpMappings.ts [hunk 3]
```diff
@@ -0,0 +1,923 @@
...
+      '{language}': translatableAttributeMapping('volunteer', {
+        value: ctx => ctx.currentValue.opportunities,
+        info: ctx => ctx.currentValue,
+        language: ctx => ctx.captures.language,
+      }),
+    },
+  },
+};
+
+export const KHP_RESOURCE_REFERENCES = {
```
Comment (stephenhand) [inline resources-domain/resources-import-producer/mappings/khpMappings.ts:691]: I don't think we need these in the lambda code do we?

**Review by stephenhand — APPROVED** [review]
LGTM, couple of nits, but gtg. I didn't go through every line of the mappings you wrote, I think we can make better use of time testing it E2E to find issues rather than me trawling through 600+ lines

Is there any reason not to hook this into the lambda execution in this PR?

**Review by stephenhand — APPROVED** [review]
LGTM

**Pull request merged**